annalieseschropp/checkers
# 10 Sound Features

This pr closes #64 #65 #67 #68. 

### What was Accomplished
* Added sound effects for:
  * Clicking a menu option
  * Moving a piece / Capturing
  * Finishing a game
* Added one custom background music track that plays non-stop
* Added sliders in the options menu to control music and SFX volume
* Linked the sliders to the respective volume controls
* Made unit tests for what I could, but manual (auditory) testing is strongly suggested.

### Testing
* To run unit tests, ensure you have a build set each scene
    * `File > Build Settings > Add Open Scenes > Build`
* You also need the Test Runner window open to see the available tests
    * `Window > General > Test Runner`
    * You can dock this where you would like based on your layout preferences
* Click `Run All` in the Test Runner window or select a specific test then click `Run Selected`

* To manually run the scene, press the `play` button or go to `File > Build and Run`
  * Check that button presses (excluding dropdowns) have a sound effect
  * Check that pieces have randomized movement sounds
  * Check that a short sound plays after a game is finished
  * Ensure that the music and SFX volume controls in the options menu actually work

### To Dos:
* Add more options to the options menu as they are needed.
